CRM/Admin/Form/ScheduleReminders.php

```diff
@@ -330,10 +330,10 @@ private static function normalizeFormValues(array $values): array {
 
     // Absolute or relative date
     if ($values['absolute_or_relative_date'] === 'absolute') {
-      $values['start_action_offset'] = $values['start_action_unit'] = $values['start_action_condition'] = $values['start_action_date'] = NULL;
+      $values['start_action_offset'] = $values['start_action_unit'] = $values['start_action_condition'] = $values['start_action_date'] = '';
     }
     else {
-      $values['absolute_date'] = NULL;
+      $values['absolute_date'] = '';
```

Comment:
Just on the `absolute_date` field, I'm wondering if this will cause some other weirdness since depending on SQL_MODE and db vendor and whether it's a Tuesday sometimes `''` will insert `0000-00-00` which is then treated differently from null (so e.g. [here](https://github.com/civicrm/civicrm-core/blob/442354eb128647184c79fa7d709f727ca0a83e75/CRM/Admin/Form/ScheduleReminders.php#L277) might get confused). I can't reproduce such a problem for this field, but it's not Tuesday, and I'm only checking mariadb 10.6 at the moment.

As much as I choke on my keyboard while typing it, perhaps the right value for absolute_date is string `'null'`. That seems to insert a real null regardless of mode.

Otherwise at least line 333 looks good, and I agree _some_ change on line 336 would fix it.

Reply:
Actually I see writeRecord converts `'null'` to `''` anyway, so if there were a problem it wouldn't be just here. Let's go with this.
...And I'll recheck on Tuesday :)